lib/pdk/validators/base_validator.rb

```diff
@@ -44,19 +46,26 @@ def self.invoke(report, options = {})
         return 0 if targets.empty?
 
         PDK::Util::Bundler.ensure_binstubs!(cmd)
-        cmd_argv = parse_options(options, targets).unshift(cmd_path)
-        cmd_argv.unshift('ruby', '-W0') if Gem.win_platform?
 
-        command = PDK::CLI::Exec::Command.new(*cmd_argv).tap do |c|
-          c.context = :module
-          c.add_spinner(spinner_text)
-        end
+        targets = (self::INVOKE_STYLE == :per_target) ? targets.combination(1).to_a : Array[targets]
```

Comment:
Yeah, this is to try and avoid duplicating the logic below when handling the different invoke styles. If `:per_target`, we take the array of targets and using `Array#combination`, split it into a 2 dimensional array, where each element inside `targets` is itself an array of 1 target.

Given
```ruby
targets = [
  'target1',
  'target2',
  'target3',
]
```
If the `INVOKE_STYLE` is `:per_target`, the `targets` array becomes `[['target1'], ['target2'], ['target3']]`. If the `INVOKE_STYLE` is `:once`, the `targets` array becomes `[['target1', 'target2', 'target3']]`.

By doing it this way, we ensure that `targets` will always be an array of arrays and just loop through it and avoid having to put a conditional around the execution logic to determine if how many times the validated should be invoked and with which targets.